### ISSUE TITLE
[Master and only ticket] Language of Parts

### ISSUE THREAD

**Issue by saad-mo** [SAP/spartacus#11439]
## Overview

- **MASTER** Ticket #11672
- [**Github label:**](https://github.com/SAP/spartacus/issues?q=is%3Aissue+is%3Aopen+label%3Aa11y-sr%2Flanguage-of-parts) `a11y-sr/language-of-parts`

Screen Reader should read out the native language 'part'. For example, a German phrase in an English sentence should be read out in German. So far only need to address this on the **Find a Store page.** 

## Implementation

- For demonstration purposes, 2 languages **different** from "defaults" will be enough. 
E.g. Spanish `lang="es"` and Polish `lang="pl"`.
- Please discuss this with Bill. Should we **deliver** it, **or provide documentation** for the customer to do it. 
- Example in Spartacus: 
https://spartacus-demo.eastus.cloudapp.azure.com/electronics-spa/en/USD/store-finder/country/DE

<img width="262" alt="Screen Shot 2021-03-09 at 11 04 54 AM" src="https://user-images.githubusercontent.com/15215400/110509547-d4aeba00-80c7-11eb-8491-6b18507a0963.png">

**Comment by Zeyber** [comment]
While the backend doesn’t actually provide any information about the language itself, it does return information based on the language you request (eg. getting product details in german). In other instances, we get translation keys which we have full control over from the front end.

It is hard to say for edge cases where some component somewhere doesn’t work. We could split the work into smaller tasks to cover the obvious cases first.

For example, assuming the language is set to German:
- Append lang to the containing element of the key that has used our translation service. It can result in lang="de" where the correct key is used and in lang="en" in case of fallback.
- In cases where the backend returns plain text (ie. not a translation key), we can assume that the returned text is in the correct language and append lang="de" based on the request. This could be more complicated as we may need to do it per component. `Note: We should not need to do this in Spartacus if we are not using any fallback language`.
- Store finder addresses are one of those edge cases. We don’t have any information from the backend which language the address will be returned in. We can only assume it will be the local language and return it based on the country. To do that, we’d also need to map country codes to native language codes. (eg. nz => en).
![image](https://user-images.githubusercontent.com/11735817/157894933-34e6e427-0f88-488b-908b-3bf2b83845b1.png)
- Other edge cases like the store finder address could also appear during the transition.
